src/common/useCustomCursor.ts

```diff
@@ -0,0 +1,26 @@
+import * as React from 'react';
+
+export default function useCustomCursor(
+    ref: React.RefObject<HTMLElement> | null,
+    cursorTemplate: string,
+    color: string,
```

Comment:
Different icons may have different color needs (fill, stroke, text, etc.). Is it premature to make this a generic component/hook?